engine/src/main/java/com/google/android/fhir/FhirEngine.kt

```
@@ -83,6 +83,17 @@ interface FhirEngine {
   @Throws(ResourceNotFoundException::class)
   suspend fun get(type: ResourceType, id: String): Resource
 
+  /**
+   * Loads multiple FHIR resources given [ResourceType] and logical IDs.
+   *
+   * @param type The type of the resource to load.
+   * @param ids The logical IDs of the resources.
+   * @return The list of requested FHIR resources.
+   * @throws ResourceNotFoundException if the resources are not found.
+   */
+  @Throws(ResourceNotFoundException::class)
+  suspend fun get(type: ResourceType, vararg ids: String): List<Resource>
```

update the exception to say that if any resources are not found, this will throw.

---

Does this mean if a request is made for 3 resources and only 2 are available it should throw an exception?

---

Is there a scenario where getting all resources in the list is mandatory and returning a subset may break functionality?